Emby.Server.Implementations/Library/IgnorePatterns.cs

```diff
@@ -0,0 +1,73 @@
+using System.Linq;
+using DotNet.Globbing;
+
+namespace Emby.Server.Implementations.Library
+{
+    /// <summary>
+    /// Glob patterns for files to ignore
+    /// </summary>
+    public static class IgnorePatterns
+    {
+        /// <summary>
+        /// Files matching these glob patterns will be ignored
+        /// </summary>
+        public static readonly string[] Patterns = new string[]
+        {
+            "**/small.jpg",
+            "**/albumart.jpg",
+            "**/*sample*",
+
+            // Directories
+            "**/metadata/**",
+            "**/ps3_update/**",
+            "**/ps3_vprm/**",
+            "**/extrafanart/**",
+            "**/extrathumbs/**",
+            "**/.actors/**",
+            "**/.wd_tv/**",
+
+            // WMC temp recording directories that will constantly be written to
+            "**/TempRec/**",
+            "**/TempSBE/**",
+
+            // Synology
+            "**/eaDir/**",
+            "**/@eaDir/**",
+            "**/#recycle/**",
+
+            // Qnap
+            "**/@Recycle/**",
+            "**/.@__thumb/**",
+            "**/$RECYCLE.BIN/**",
+            "**/System Volume Information/**",
+            "**/.grab/**",
+
+            // Unix hidden files and directories
+            "**/.*/**",
```

Comment:
does this also handle hidden files in non-hidden directories?

Reply:
Yes, and there is a test for this